internal/commands/app/delete.go

```diff
@@ -0,0 +1,80 @@
+package app
+
+import (
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+
+	"github.com/spf13/pflag"
+)
+
+// CommandDelete is the `app delete` command
+type CommandDelete struct {
+	inputs      cli.ProjectInputs
+	realmClient realm.Client
+	outputs     []appOutput
+}
+
+// Flags is the command flags
+func (cmd *CommandDelete) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+}
+
+// Setup is the command setup
+func (cmd *CommandDelete) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = profile.RealmAuthClient()
+	return nil
+}
+
+// Handler is the command handler
+func (cmd *CommandDelete) Handler(profile *cli.Profile, ui terminal.UI) error {
+	apps, appErr := cli.ResolveApps(ui, cmd.realmClient, cmd.inputs.Filter())
+	if appErr != nil {
+		return appErr
+	}
+
+	for _, app := range apps {
+		err := cmd.realmClient.DeleteApp(app.GroupID, app.ID)
+		cmd.outputs = append(cmd.outputs, appOutput{app: app, err: err})
+	}
+	return nil
+}
+
+// Feedback is the command feedback
+func (cmd *CommandDelete) Feedback(profile *cli.Profile, ui terminal.UI) error {
+	if len(cmd.outputs) == 0 {
+		return ui.Print(terminal.NewTextLog("No apps to delete"))
+	}
+
+	var logs []terminal.Log
+	logs = append(logs, terminal.NewTableLog(
+		"Deleted app(s)",
```

Comment:
I originally wanted to have text indicating the number of apps we deleted, but without going through the output and  counting how many errors we got, we can't put that number here. I am fine leaving as is or storing a count in a variable (which is simple once this is all done in Handler)

Reply:
first off, you should rebase and repurpose this `Feedback` function into the `Handler` itself

then, while you're looping through and deleting apps, you could keep an incrementing count of non-nil errors there

---

fwiw, i am starting to feel like this table output too verbose/heavy-handed, but i also don't want to go in and make cli-wide changes to that yet.  i say this here just so you dont go crazy over-architecting a feedback message here when we may be simplifying it all in the first place

in the context of someone running `realm-cli apps delete --app a1 --app a2 ...`, i think the following output would be sufficiently clear:

```
Successfully deleted 2 apps
```
or
```
Successfully deleted 1 / 2 apps
Failed to delete app: a2
```
or
```
Successfully deleted 0 / 2 apps
Failed to delete apps:
  - a1
  - a2
```

to be clear, we should not be implementing this in this ticket, just leave this consistent with our other delete methods for now.  but avoid making changes to any other commands (like `app list` here for now)